deploy-platform/src/github-action/Inputs.ts

```diff
@@ -1,18 +1,45 @@
 import * as core from "@actions/core";
 
-export function getInputs() {
+import { DeployInputs } from "../lib/Deploy.types";
+
+export function getInputs(): DeployInputs {
   const project = core.getInput("project", { required: true });
   const token = core.getInput("token", { required: true });
   const environment = core.getInput("environment", { required: true });
   const repoName = core.getInput("repo-name");
   const workflowId = core.getInput("workflow-id");
   const ref = core.getInput("ref");
-  return {
+  const requestInterval = core.getInput("request-interval");
+  return sanitizeInputs({
     project,
     token,
     environment,
     repoName,
     workflowId,
     ref,
-  };
+    requestInterval,
+  });
+}
+
+function sanitizeInputs(inputs: ActionInputs): DeployInputs {
+  if (inputs.requestInterval) {
+    const requestInterval = parseInt(inputs.requestInterval);
+    if (isNaN(requestInterval)) {
+      throw new Error("Input request-interval must be a number");
+    } else {
+      return { ...inputs, requestInterval };
+    }
+  } else {
+    return { ...inputs, requestInterval: undefined };
+  }
```

Comment:
Why are you explicitly setting `requestInterval` to undefined? I suppose that it is because github actions give you a value of empty string when it is not defined. But here it is very strange to see this. So, maybe it would be more understable if you change the comparison to a difference of empty string. Or even better, simply pass the empty string, because I can see in the config that it is compatible with it.

```suggestion
  if (inputs.requestInterval !== "") {
    const requestInterval = parseInt(inputs.requestInterval);
    if (isNaN(requestInterval)) {
      throw new Error("Input request-interval must be a number");
    }
    return { ...inputs, requestInterval };
  }
  return inputs;
```

Reply:
Solved